dask/dataframe/io/tests/test_io.py

```diff
@@ -679,6 +684,22 @@ def test_from_delayed():
     assert str(e.value).startswith("Metadata mismatch found in `from_delayed`")
 
 
+def test_from_delayed_optimize_fusion():
+    # Test that DataFrame optimization fuses a `from_delayed`
+    # layer with other Blockwise layers and input Delayed tasks.
+    # See: https://github.com/dask/dask/pull/8852
+    ddf = (
+        dd.from_delayed(
+            map(delayed(lambda x: pd.DataFrame({"x": [x] * 10})), range(10)),
+            meta=pd.DataFrame({"x": [0] * 10}),
+        )
+        + 1
+    )
+    # NOTE: Fusion requires `optimize_blockwise`` and `fuse_roots`
```

Comment:
This is a great point. I had honestly forgotten that `fuse_roots` will return a materialized Layer, but that makes sense. Although we may be able to rewrite fuse roots to avoid materialization, we may not get much out of that. What we really want is the graph to remain HLG when `dsk.cull` is called. So, perhaps we just want to move the cull operation to occur in between `optimize_blockwise` and `fuse_roots`?  I'll throw up a small separate PR to check if this idea breaks anything :)